components/sup-protocol/build.rs

```diff
@@ -38,9 +38,10 @@ fn generate_protocols() {
           .expect("protocols");
     compile_proto_impls(&protocol_files(), &protocol_includes()).expect("protocol-impls");
     for file in generated_files() {
-        fs::rename(&file,
-                   format!("src/generated/{}",
-                           file.file_name().unwrap().to_string_lossy())).unwrap();
+        fs::copy(&file,
+                 format!("src/generated/{}",
+                         file.file_name().unwrap().to_string_lossy())).unwrap();
+        fs::remove_file(&file).unwrap();
```

Comment:
Why is this change necessary? It seems like we could end up failing to do the `remove_file` if the `copy` fails, and then I'm not sure what the outcome would be for the build script if we have stale files sitting around.

Reply:
The `expect sounds good. Windows cannot rename a file across different drives. The error emitted by rename is `The system cannot move the file to a different disk drive.`

Reply:
Oh! The rename doesn't work *across* mounts. I get it now. But what I don't understand is why this *is* a move across mounts. Why are the files generated in one mount and then moved to another? It looks like we can choose where we do the generation, so why not do it within the same mount?

Reply:
Thats a good point. It was generating them in `OUT_DIR`. I just changed it to generate directly to `src/generated` and it works fine and also removes several more lines of code which is always a win!